Userland/Libraries/LibWeb/Layout/BlockContainer.cpp

```diff
@@ -26,6 +27,13 @@ Painting::PaintableWithLines const* BlockContainer::paintable_with_lines() const
     return static_cast<Painting::PaintableWithLines const*>(Box::paintable_box());
 }
 
+void BlockContainer::prepare_for_replaced_layout(){
+    if (dom_node() && dom_node()->is_html_input_element()) {
```

Comment:
I wonder: is this correct for non-text input types? Like `type=color` or `type=file` or `type=button`?

Reply:
I guess this `if` statement has to match the one in `box_is_sized_as_replaced_element`, but I think most of those other types don't use the `size` attribute. Though I notice that the existing code _did_ apply to all types (except `file`? maybe we need to exclude that?) so maybe there's no added bug here even if it still looks wrong.